ledger/accountdb.go

```diff
@@ -245,42 +343,69 @@ func makeCompactAccountDeltas(accountDeltas []ledgercore.AccountDeltas, baseAcco
 }
 
 // accountsLoadOld updates the entries on the deltas.old map that matches the provided addresses.
-// The round number of the persistedAccountData is not updated by this function, and the caller is responsible
+// The round number of the dbAccountData is not updated by this function, and the caller is responsible
 // for populating this field.
-func (a *compactAccountDeltas) accountsLoadOld(tx *sql.Tx) (err error) {
-	if len(a.misses) == 0 {
+func (c *compactAccountDeltas) accountsLoadOld(tx *sql.Tx) (err error) {
+	if len(c.misses) == 0 {
 		return nil
 	}
 	selectStmt, err := tx.Prepare("SELECT rowid, data FROM accountbase WHERE address=?")
 	if err != nil {
 		return
 	}
 	defer selectStmt.Close()
+	loadStmt, err := tx.Prepare("SELECT data FROM accountext WHERE id=?")
+	if err != nil {
+		return
+	}
+	defer loadStmt.Close()
```

Comment:
I think that lazy initialization here would be better since on the "normal case", I don't expect to see too many accounts that have many holdings.